prime-router/docs/features/0005-managing-translation-schemas.md

```diff
@@ -0,0 +1,765 @@
+
+#### HL7 -> FHIR
+Long term, it would be great to add support to this library (it's open sourced) that would support reading files from different file-like storage solutions rather than just disk, but this would likely not be feasible to get done in a quick enough timeframe.  Instead, before instantiating the converter, the application will sync the azure storage to a spot on the disk the library will read from.
+
+#### Cache
+**This could potentially be deferred until it's been identified that the network requests to azure blob storage are the bottleneck**
+
+Since the pipeline will now require reading several files out of the blob store at multiple spots, it will likely make sense to cache the resolved translation schemas since they will change relatively infrequently.  This could initially be done with a simple in-memory cache using the translation schema URI as the cache key.
+
+#### Pros
+- Maintains the current file based model for the translation schemas
+- Translation schemas can be easily edited by hand via the Azure UI
+- Supports reading translation schemas from azure as well as from disk
+    - Enables leaving the common translation schemas on disk if that's preferred
+- Enables easy rollback
+    - Post MVP, a UI could be enabled to allow selecting what version to go back to
+- Supports reading files from azure or from disk
+
+#### Cons
+- Need to support to different URI schemes
+- Relies on pulling files out of azure which is slower than disk or the database
+- Requires more complicated error handling
+
+## Managing translation schemas
+The translation schemas break down into two categories:
+
+- translation schemas dedicated to a specific sender or receiver
+  - one of them will typically be set in the `setting` for either the sender or receiver
+- translation schemas that are used across multiple senders or receivers across organizations
+    - for now, these will continue to be stored in the source code since changes here should be carefully rolled out
+    - an example of this would be `ADT-01` directory
+
+### Directory Structure
+
+Below is the directory structure that we would implement inside the azure blob store.  It includes the location of
+where common translation schemas used across organizations would go, but the current thought is whether (and which ones) they should exist in the blob
+store will be decided down the road since this design supports the file living in either location.  The largest change from the existing
+directory structure is that a sample input and output file will be included in order to verify the translation schema works as expected.
+
+**Note: the `fhir_mapping` directory will be lifted exactly as is to azure since the directory structure is decided by the underlying library**
+
+```
+/azure-blob-container
+    /fhir_mapping
+        valid-2023-03-29T23:53:00.488Z.txt
```

Comment:
This is confusing to me, each schema should have its own input/output test file, no? OR at least its own output test file?

Reply:
Each one does... but because of `extend` our validity exists for all the schemas collectively, not an individual basis.